### PR TITLE
Implemented TranspOtter modifications

### PR DESCRIPTION
**Pull request by Mezgrman**
Ideally, just `#define TRANSPOTTER` to build a TranspOtter firmware with FOC.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -52,7 +55,7 @@ startup_stm32f103xe.s
 #######################################
 # binaries
 #######################################
-PREFIX = arm-none-eabi-
+PREFIX = /usr/local/gcc-arm-none-eabi-5_4-2016q3/bin/arm-none-eabi-
```
Comment (LeoDJ) [inline Makefile]: That line should probably be changed back

**Comment by EmanuelFeru** [comment]
* Merged with latest commit
* Fixed conflicts

**Open points:**
1. I see that you define 
  #define SPEED_COEFFICIENT   14746  // 0.9f - higher value == stronger. 0.0 to ~2.0?
  #define STEER_COEFFICIENT   8192   // 0.5f - higher value == stronger. if you do not want any steering, set it to 0.0; 0.0 to 1.0

2. In config.h but they are not used for TRANSPOTTER. Do we need them?

**Nice to have:**
* I see that there are some floating-point calculation for TRANSPOTTER. Would be nice to convert all of them to fixed-point for execution speed and make the firmware floating-point free again.

**Comment by Mezgrman** [comment]
I will have to check, since I didn't write the code myself, I just copied the existing TranspOtter code. I can check if we need SPEED_COEFFICIENT and STEER_COEFFICIENT. I don't know why the main loop is faster for TranspOtter either. Can you check the floating point thing? I wasn't sure how to proceed here.

**Comment by EmanuelFeru** [comment]
1. It would be great if you can check how does it behave if we run the main loop at the default 5 ms  sample time. Can you check?
2. Regarding the coefficients we can keep them in, but I would recommend to comment them-out because they are not used for TRANSPOTTER.
3. Regarding the floating-point conversion to fixed-point I will do that later, because I need to understand the calculations

**Comment by Mezgrman** [comment]
I will check that soon, hopefully until Wednesday!

**Comment by EmanuelFeru** [comment]
Merge done. Thanks a lot for the effort!

**Pull request merged**